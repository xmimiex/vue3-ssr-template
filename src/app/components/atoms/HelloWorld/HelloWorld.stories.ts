import { Meta, Story } from '@storybook/vue3'
import HelloWorld from './HelloWorld.vue'

export default {
  title: 'Atoms/HelloWorld',
  argTypes: {
    msg: {
      control: { type: 'text', defaultValue: 'Hello World' },
    },
  },
} as Meta

const Template: Story = (args: any) => ({
  components: { HelloWorld },
  setup() {
    return { args }
  },
  template: `
    <HelloWorld v-bind="args" />`,
})


export const Default = Template.bind({})
Default.args = {
  msg: 'Hello world',
}