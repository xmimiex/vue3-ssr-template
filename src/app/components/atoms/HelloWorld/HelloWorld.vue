<template>
  <div class="hello">
    <h1>{{ msg }}</h1>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
  name: 'HelloWorld',
  props: {
    msg: String,
  },
})
</script>

<style scoped lang="scss">
  h3 {
    margin: 2rem 0 0;
  }

  li {
    display: inline-block;
    margin: 0 1rem;
  }

  a {
    color: #42b983;
  }
</style>
