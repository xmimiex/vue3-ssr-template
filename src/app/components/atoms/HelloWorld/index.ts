export { default } from './HelloWorld.vue'
