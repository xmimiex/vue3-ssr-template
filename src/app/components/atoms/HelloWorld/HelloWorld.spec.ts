import { shallowMount } from '@vue/test-utils'
import HelloWorld from '.'

describe('HelloWorld', () => {

  it('should match snapshot', () => {
    const wrapper = shallowMount(HelloWorld, {
      propsData: {
        msg: 'World',
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
