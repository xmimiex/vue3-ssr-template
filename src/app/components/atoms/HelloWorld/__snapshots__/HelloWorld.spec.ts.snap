// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld should match snapshot 1`] = `
<div
  class="hello"
>
  <h1>
    World
  </h1>
</div>
`;
